### PR TITLE
feat(AwsRedisInstance): make authEnabled mutable

### PR DESCRIPTION
**Pull request by dushanpantic**
**Description**

Changes proposed in this pull request:
- makes `authEnabled` mutable

Required flow to disable auth on redis elasticache is described in AWS docs here: https://docs.aws.amazon.com/AmazonElastiCache/latest/red-ug/in-transit-encryption-disable.html

**Related issue(s)**
https://github.com/kyma-project/cloud-manager/issues/580

### CODE REVIEW

File: api/cloud-control/v1beta1/redisinstance_types.go [hunk 1]
```diff
@@ -211,6 +211,7 @@ type RedisInstanceAzure struct {
 	ReplicasPerPrimary int `json:"replicasPerPrimary,omitempty"`
 }
 
+// +kubebuilder:validation:XValidation:rule=(self.authEnabled == false || self.transitEncryptionEnabled == true), message="authEnabled can only be true if TransitEncryptionEnabled is also true"
```
Comment (ijovovic) [inline api/cloud-control/v1beta1/redisinstance_types.go:214]: I am not sure this validation rule would work as expected? Eg, authEnabled = false and transitEncryptionEnabled = true will trigger it, but the message will not make sence.

File: pkg/kcp/provider/aws/redisinstance/deleteUserGroup.go [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+	userGroupState := ptr.Deref(state.userGroup.Status, "")
+	if userGroupState == awsmeta.ElastiCache_UserGroup_DELETING {
+		return nil, nil
+	}
+
+	logger.Info("Deleting userGroup")
+
+	err := state.awsClient.DeleteUserGroup(ctx, *state.userGroup.UserGroupId)
+	if err != nil {
+		return awsmeta.LogErrorAndReturn(err, "Error deleting userGroup secret", ctx)
```
Comment (ijovovic) [inline pkg/kcp/provider/aws/redisinstance/deleteUserGroup.go:29]: remove the work 'secret' from the message

File: api/cloud-control/v1beta1/redisinstance_types.go (hunk 1 above)
Comment (dushanpantic) [reply api/cloud-control/v1beta1/redisinstance_types.go:214]: It will not trigger it

Comment (dushanpantic) [reply api/cloud-control/v1beta1/redisinstance_types.go:214]: `self.authEnabled`==false will be satisfied

**Review by ijovovic — APPROVED** [review]
lgtm

**Pull request merged**